test/controllers/api_users/permissions_controller_test.rb

```diff
@@ -308,6 +308,23 @@ class ApiUsers::PermissionsControllerTest < ActionController::TestCase
       end
     end
 
+    should "prevent access if the user does not have an access token for the application" do
```

Comment:
The commit message mentions "in addition to having an expired one" - I think we only have tests for revoked tokens, not expired ones? Do we need one for expired tokens? Do these tokens expire?

Reply:
I've done a bit of searching for expired tokens in test code - we don't test them elsewhere much - the only thing I saw was us testing them in the `access_token_test`. I tried adding a test setting an access token as `expired` but it didn't raise an error - I may not have been setting this up properly though.

I'd like to investigate this further but will do so outside the scope of this PR!